src/state/reducers/companion_windows.js

```diff
@@ -0,0 +1,65 @@
+import ActionTypes from '../actions/action-types';
+
+/**
+ * getCompanionWindowForPosition - Finds the companion window with the given windowId and position
+ * This will not be necessary once we have multiple companion windows in a single position and
+ * we can convert this and SET_COMPANION_WINDOW into a simple ADD_COMPANION_WINDOW reducer
+ */
+function getCompanionWindowForPosition(state, windowId, position) {
+  return Object.values(state).find(cw => (
+    cw.windowId === windowId && cw.position === position
+  ));
+}
+
+/**
+ * companionWindowsReducer
+ */
+export const companionWindowsReducer = (state = {}, action) => {
+  switch (action.type) {
+    // action params: id
+    case ActionTypes.CLOSE_COMPANION_WINDOW:
```

Comment:
I don't understand why the companion window will not be removed entirely on CLOSE_COMPANION_WINDOW. Is there a wanted UI behavior that makes that necesarry?

Reply:
Closing it (as opposed to removing it all-together) allows the user's scroll position to be retained when re-opening.  That's useful for a single companion window, but won't necessarily make sense once we have multiple companion windows in a particular position.  I'll change this to removing the companion window from state outright so that it doesn't cause any confusion down the road.

Reply:
Ok, I see :+1:

Reply:
Just pushed up the change to replace `CLOSE_COMPANION_WINDOW` with `REMOVE_COMPANION_WINDOW` behavior.  Thanks for pointing that out.